arangod/Replication2/StateMachines/Document/DocumentLeaderState.cpp

```diff
@@ -511,6 +511,15 @@ auto DocumentLeaderState::modifyShard(ShardID shard, CollectionID collectionId,
                                           std::move(origin));
   if (trxLock.fail()) {
     return trxLock.result();
+  } else if (trxLock.get() == nullptr) {
+    TRI_ASSERT(false) << "Locking shard " << shard
```

Comment:
Currently, never. That was just my "defensive" programming 🤓
With the unique_ptr being out of this function's control, better safe then sorry, in case someone modifies the `lockShard()` method and accidentally forgets about this place, at least there's no UB.